api/envoy/api/v2/core/health_check.proto

```diff
@@ -95,12 +104,19 @@ message HealthCheck {
     repeated string request_headers_to_remove = 8;
 
     // If set, health checks will be made using http/2.
-    bool use_http2 = 7;
+    // Deprecated, use :ref:`codec_client_type <envoy_api_field_core.HealthCheck.HttpHealthCheck.codec_client_type>` instead.
+    bool use_http2 = 7 [deprecated = true];
```

Comment:
This should be documented in docs/root/intro/deprecated.rst

Reply:
done